### ISSUE TITLE
Changing disk UUID after disk replacing in xcp-ng with Xen Orchestra

### ISSUE THREAD

**Issue by top-secrett** [apache/cloudstack#7957]
##### ISSUE TYPE

 * Improvement Request

##### CLOUDSTACK VERSION

~~~
Any
~~~

##### SUMMARY

Hello. I use CloudStack with XCP-NG and I am in search of the best backup solution.
All software, including Xen Orchestra, can restore a backup as new VM with a new disk UUID. After that I detach disk from original VM and attach the recovered disk. VM is working fine, but CloudStack cannot manage the disk due to the new disk UUID.
If I replace disk UUID in CloudStack databases, disk management will be performed normally
Example:
~~~
update volumes SET path="24c34f4d-35b3-400a-92e3-c27ba154aabf" where id=2283;
~~~
Maybe in new version of cloudstack you can add function of manually or automatically changing disk UUID from Cloudstack?
Or import VM from XCP-NG function.

**Comment by top-secrett** [comment]
@neil3333 yes, it is a promlem too. I think you should open new issue

**Comment by DaanHoogland** [comment]
this sounds like you want to write a plugin for the backup and restore framework.

Normally a VM uuid would not change in cloudstack. What may happen is that you get a new VM in Xen, but in the cloudstack DB it should not have changed.

**Comment by top-secrett** [comment]
@DaanHoogland after stop and start VM in CS, VM's uuid changes in Xen. All backup solutions for xcp-ng cannot backup VM after that. It is a big problem, because there is no any full compatible backup sofy for xcp-ng with CS

**Comment by DaanHoogland** [comment]
> @DaanHoogland after stop and start VM in CS, VM's uuid changes in Xen. All backup solutions for xcp-ng cannot backup VM after that. It is a big problem, because there is no any full compatible backup soft for xcp-ng with CS

I understand, so these backup solutions run out of bounds from cloudstack, right? cloudstack is unaware and on start recreates the VM (this is how it always works on xen/kvm) What is your backup solution?

**Comment by DaanHoogland** [comment]
> @DaanHoogland We use XenOrchestra and Vinchin

/me googling ;)

**Comment by top-secrett** [comment]
@DaanHoogland should I open new issue?

**Comment by DaanHoogland** [comment]
> @DaanHoogland should I open new issue?

no, I would refine the issue here to be a better specification. We don't need an extra issue. What do we need, either
- adding support to the B&R framework for Vinchin
- dealing with VM restarts (possibly controlled by a setting)

ACS is not aware of out of bounds backup solution and making it aware is making them no longer out of bounds. We (sd in you) need to define a good functional design to how this should work. Next we can decide if it is worth the effort. I would not call it an improvemnt request depending on the way forward. Are you familiar with the B&R framework?

**Comment by top-secrett** [comment]
@DaanHoogland Vinchin didnt find any solution. Perhaps in a next release you will add feature that disables VM reconfiguration?

**Comment by DaanHoogland** [comment]
> @DaanHoogland Vinchin didnt find any solution. Perhaps in a next release you will add feature that disables VM reconfiguration?

If we can figure out how to do that in a clean way, sure.  However disks/volumes are first class citizens in cloudstack. allowing a new disk to have the same UUID as  prior one is a no go. I am not sure if this can be fitted into the current B&R design (cc @rohityadavcloud ?)

**Comment by top-secrett** [comment]
@DaanHoogland I meant feature that disables VM reconfiguration to prevent generating a new VM UUID after reboot.

**Comment by DaanHoogland** [comment]
@top-secrett , that is a similar issue. As ACS installs a new VM based on the old disk when restarting an old VM. We can see if we can and how to deal with that.

**Comment by DaanHoogland** [comment]
@top-secrett , are you still looking at this? it is stale, should we close or save as long term plan/future work?
